javascript/javascript3/week1/weatherApp/weatherApp.js

```diff
@@ -0,0 +1,125 @@
+//Sign up for api key
+//First call to the weather api
+//Fetch weather data from a city
+const txtcityName = document.querySelector('#txtCityName');
+const fetchButton = document.querySelector('#fetchButton');
+const appKey = '6e92755c3cd985413c44c7aac6eff7ec';
+
+const getWeatherData = () => {
+    
+    const cityName = txtcityName.value;
+    if (!cityName) {
+        alert('Provide a city name!');
+    }
+    else {
+        
+        fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${appKey}`).then(res => res.json()).then((data) => {
+            if ( data.value == 0){
+                alert('No weather data found!')
+            }
+            else {
+                clearOldData();
+                displayWeatherData(data);
+            }
+        })
```

Comment:
It would be nice if the app had some simple error handlers to display some error messages for the users. So when for example the user enters a city name that doesn't exist or if the api is not responding.

You can do that by checking the `data` variable (which is the response from the server).
It will return this object in that case (which you can check and display to the user).

![Screenshot 2021-07-01 at 12 32 00](https://user-images.githubusercontent.com/28038265/124110300-5e640800-da68-11eb-9265-a233570067c6.png)

you will learn more about it in NodeJS module :) 

Or, you can add a error handler function in the `catch()` method, so if anything goes  wrong when fetching data the `catch()` method wil be triggered.

You wil also learn more about it when you take `Promises` in week 2.

Other than that, it looks good!